### PR TITLE
Devon

### PR DESCRIPTION
**Pull request by SapceDandy**
Added base code for the home screen and a pop-up signing screen. I also implemented a test for signing and signup. The main branch code was merged with my branch, but they only slightly differ.

### CODE REVIEW

File: .github/workflows/Jest.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Jest
```
Comment (Alisa007) [inline .github/workflows/Jest.yml:1]: we already have jest in ci, why did you add it again?

File: .github/.keep
Comment (Alisa007) [inline .github/.keep:1]: why did you add keep file?

File: code/src/app/globals.css [hunk 2]
```diff
@@ -2,6 +2,16 @@
 @tailwind components;
 @tailwind utilities;
 
+html,
```
Comment (Alisa007) [inline code/src/app/globals.css:5]: default behavior of scrollbar would be better UX here

File: code/src/components/MainPage.tsx [hunk 3]
```diff
@@ -1,30 +1,530 @@
...
+              Terms & Agreements
+            </p>
+            <p className="w-1/3 text-sm text-center text-custom-grey">
+              @2023 RentalCompany All Rights Reserved
+            </p>
+            <p className="w-1/3 text-sm text-center text-custom-grey">
+              Privacy Policy
+            </p>
+          </div>
+        </footer>
```
Comment (Alisa007) [inline code/src/components/MainPage.tsx:526]: The component should consist of 500 lines of code, anything over 200 is very bad. Need to split it up and try to reuse code. Other components are also too long

File: code/src/components/SignIn.spec.tsx [hunk 4]
```diff
@@ -9,19 +9,33 @@ jest.mock('next/navigation', () => ({
...
-      <SignIn
-        email={undefined}
-        setEmail={() => {}}
-        password="password"
-        setPassword={() => {}}
-      >
-        children
-      </SignIn>,
+  test('should render login form when status is true', () => {
+    const { getByRole, getByPlaceholderText } = render(<SignInForm status />);
```
Comment (Alisa007) [inline code/src/components/SignIn.spec.tsx:13]: If you want to test SignInForm you should do it in SignInForm spec, it's a different component

**Comment by Alisa007** [comment]
You copy pasted old version of AccessPage component, you should just use a new version of AccessPage

File: code/src/components/MainPage.tsx (hunk 3 above)
Comment (Alisa007) [reply code/src/components/MainPage.tsx:526]: You copy pasted old version of AccessPage component, you should just use a new version of AccessPage instead of your 500 lines file and update it as you need

File: .github/.keep
Comment (SapceDandy) [reply .github/.keep:1]: I just pulled the code in the main file and merged it into my branch.

File: .github/workflows/Jest.yml (hunk 1 above)
Comment (SapceDandy) [reply .github/workflows/Jest.yml:1]: I just merged the code from the main to mine. I didn't add anything outside of the home screen code and sign-in/sign-up test.

File: demo/CS673_presentation0_team6 (video).mp4
Comment (ncruz93) [inline demo/CS673_presentation0_team6 (video).mp4:1]: Did you mean to upload this file?

File: doc/CS673_presentation0_team6.pptx
Comment (ncruz93) [inline doc/CS673_presentation0_team6.pptx:1]: Did you mean to upload this file?

**Review by KristampsWong — APPROVED** [review]

**Comment by SapceDandy** [comment]
Deleted unused files.

**Review by Alisa007 — APPROVED** [review]

**Comment by Alisa007** [comment]
Main page file shouldn't be 500 lines of code, it needs to be decomposed, code reused, scroll shouldn't be always hidden etc
